modules/i3s/src/i3s-attributes-loader.js

```diff
@@ -0,0 +1,119 @@
+import {fetchFile} from '@loaders.gl/core';
+/* global Buffer */
+/** @typedef {import('@loaders.gl/loader-utils').LoaderObject} LoaderObject */
+
+// __VERSION__ is injected by babel-plugin-version-inline
+// @ts-ignore TS2304: Cannot find name '__VERSION__'.
+const VERSION = typeof __VERSION__ !== 'undefined' ? __VERSION__ : 'latest';
+const STRING_ATTRIBUTE_TYPE = 'String';
+const OBJECT_ID_ATTRIBUTE_TYPE = 'Oid32';
+const FLOAT_64_TYPE = 'Float64';
+const FILES = 'FILES';
+const ARRAY_BUFFERS = 'ARRAY_BUFFERS';
+
+async function parseI3STileAttributes(tile, options) {
+  if (!tile.attributeUrls) {
+    return tile;
+  }
+
+  const filesWithAttributes = await getFilesOrArrayBuffers(tile.attributeUrls, FILES);
+  const arrayBuffers = await getFilesOrArrayBuffers(filesWithAttributes, ARRAY_BUFFERS);
+
+  tile.attributes = getTileAttributes(arrayBuffers, options);
+  return tile;
+}
+
+async function getFilesOrArrayBuffers(items, type) {
+  const promises = [];
+
+  for (let index = 0; index < items.length; index++) {
+    const fileOrUrl = items[index];
+
+    switch (type) {
+      case FILES:
+        promises.push(fetchFile(fileOrUrl));
+        break;
+      case ARRAY_BUFFERS:
+        promises.push(fileOrUrl.arrayBuffer());
+        break;
+      default:
```

Comment:
This is an error if you split the function you get rid of this case.

Reply:
Done